src/adapter/src/catalog/open.rs

```diff
@@ -1153,6 +1297,61 @@ impl BuiltinBootstrapClusterSizes {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct SystemViewDependenciesVisitor {
+    names: BTreeSet<UnresolvedItemName>,
+}
+
+// Visitor that records the named dependencies for a statement.
+impl<'ast> Visit<'ast, Raw> for SystemViewDependenciesVisitor {
+    fn visit_item_name(&mut self, item_name: &'ast <Raw as AstInfo>::ItemName) {
+        let RawItemName::Name(name) = item_name else {
+            return;
+        };
+        // Dependent views must have only a schema specified.
+        if name.0.len() != 2 {
+            return;
+        }
+        self.names.insert(name.clone());
+    }
+
+    fn visit_expr(&mut self, node: &'ast Expr<Raw>) {
+        // Casts to reg* types involve reading from views, so any view with these casts must also
+        // depend on those views. In typeconv.rs see STRING_REG_CAST_TEMPLATE for raw SQL. If the
+        // view dependencies there change, this must also be updatedj.
```

Comment:
What about functions? I think certain functions are implemented using other catalog views and if this view calls that function then we would need to somehow collect those catalog views.